x-pack/plugins/rule_registry/common/types.ts

```diff
@@ -302,3 +305,21 @@ export interface ClusterPutComponentTemplateBody {
     mappings: estypes.MappingTypeMapping;
   };
 }
+
+export interface BrowserField {
+  aggregatable: boolean;
+  category: string;
+  description?: string | null;
+  example?: string | number | null;
+  fields: Readonly<Record<string, Partial<BrowserField>>>;
```

Comment:
Wouldn't make more sense here to make `fields` optional and remove the `Partial`? 🤔 
`fields?: Readonly<Record<string, BrowserField>>;`